Assets/Shopify/UIToolkit/Editor/ProductPicker.cs

```diff
@@ -0,0 +1,155 @@
+﻿namespace Shopify.UIToolkit.Editor {
+    using System.Collections;
+    using System.Collections.Generic;
+    using UnityEngine;
+    using UnityEditor;
+    using Shopify.Unity;
+    using Shopify.Unity.SDK;
+    using System;
+    using System.Linq;
+
+    public interface IProductPickerView {
+        void DrawProductLoadingState(SerializedObject serializedObject);
+        void DrawProductPicker(SerializedObject serializedObject);
+        void DrawProductLoadingError(SerializedObject serializedObject);
+        void DrawShopHasNoProductsState(SerializedObject serializedObject);
+    }
+
+    public class ProductPicker : IProductPickerView {
+        private class ProductNameToGIDMapEntry {
+            public string ProductName;
+            public string ProductGID;
+        }
+
+        public IProductPickerView View;
+        private static List<ProductNameToGIDMapEntry> _cachedProductNameToGIDMap;
+        private List<ProductNameToGIDMapEntry> _productNameToGIDMap;
+        private ShopifyClient _client;
+        private ShopifyError _error;
+
+        public bool ProductsFinishedLoading() {
+            return _productNameToGIDMap != null || _error != null;
+        }
+
+        public ProductPicker(ShopifyClient client) {
+            _client = client;
+            View = this;
+            LoadProducts();
+        }
+
+        public void DrawInspectorGUI(SerializedObject serializedObject) {
+            if (_error != null) {
+                View.DrawProductLoadingError(serializedObject);
+                return;
+            }
+
+            if (!ProductsFinishedLoading()) {
+                View.DrawProductLoadingState(serializedObject);
+                return;
+            }
+
+            if (_productNameToGIDMap.Count == 0) {
```

Comment:
I wonder if it this boolean checking logic would be more solid if it was implementing using a simple state machine with EMPTY, LOADING, HAS_PRODUCTS states or something. It's probably overkill for this though.